components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -1408,6 +1481,23 @@ - (void)setContentIsTranslucent:(BOOL)contentIsTranslucent {
   _statusBarShifter.snapshottingEnabled = !contentIsTranslucent;
 }
 
+- (CGFloat)flexibleHeaderSafeAreaTop {
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    // Edge case for UITableViewController: If we're a subview of _trackingScrollView,
+    // we need to get the Safe Area insets from there and not use ours.
+    // Note: This appears to be fixed on iOS 11.1.
+    if ([self isDescendantOfView:_trackingScrollView]) {
+      return _trackingScrollView.safeAreaInsets.top;
+    } else {
+      return self.safeAreaInsets.top;
+    }
+  }
+#endif
+  // If < iOS 11.0 we should return the hardcoded status bar height for backwards compatibility.
+  return kPreIOS11ExpectedStatusBarHeight;
```

Comment:
This looks pretty similar to DeviceTopSafeAreaInset - is it time to extract that impl to MDC?